openml-utils/src/test/java/com/feedzai/util/provider/AbstractProviderModelBaseTest.java

```diff
@@ -255,6 +275,33 @@ private void assertAllInstances(final List<List<Future<Integer>>> evaluationsLis
         return results;
     }
 
+    /**
+     * Classifies one of two dummy instances using a given model in parallel using a given {@link ExecutorService}.
+     * This method creates the same number of events for each dummy instance.
+     *
+     * @param model    Model to use to classify dummy instances.
+     * @param executor Executor to use to launch the threads that will classify each instance.
+     * @return A list of futures that will return the class distribution of each instance.
+     * @since 0.2.0
+     */
+    private List<Future<double[]>> classDistTwoInstancesInParallel(final M model,
+                                                                   final ExecutorService executor) {
+        final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
I think a [cyclic barrier](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CyclicBarrier.html) would be semantically more correct here, but this seems fine also